openshiftcontrolplane/v1/types.go

```diff
@@ -46,6 +46,20 @@ type OpenShiftAPIServerConfig struct {
 
 	// TODO this needs to be removed.
 	APIServerArguments map[string][]string `json:"apiServerArguments"`
+
+	// apiServers holds information about enabled/disabled API servers
+	ApiServers ApiServers `json:"apiServers"`
+}
+
+type ApiServers struct {
+	// enabled is a list of enabled API servers
+	// if empty all controllers are enabled
+	// must not collide with the list of disabled API servers
+	Enabled []string `json:"enabled,omitempty"`
```

Comment:
If I were reading this godoc via `oc explain` how would I know what the valid options to set in this and the disabled list are? What's the format of the controller names? Where would I find the available options? Is there a standard format for these?

Can we add kubebuilder tags? I know they aren't used, but I think, when authoring the tags, it can help a reviewer understand what the usage of this is, to help understand what documentation should be there.

Eg things like optional vs required, min/max length of these lists, behaviour when emitted (which you've already done).

Reply:
Also, what happens if I were to put an invalid value into this? Would be good to include the available controllers if that is a finite list that we expect will only change on a rare occasion

Reply:
> If I were reading this godoc via oc explain

This type does not exist in kube-apiserver. It's used only for (de)serialization and meant exclusively for the openshift-apiserver and its operator.

> What's the format of the controller names? Where would I find the available options? Is there a standard format for these?

I was hoping to reduce the usage only for the internal purposes. Is there any convention for internal use only purposes? Until now there was no list of available apiservers to configure. I am creating the list as follows:
```
- "openshift.io/apps"
- "openshift.io/authorization"
- "openshift.io/build"
- "openshift.io/image"
- "openshift.io/project"
- "openshift.io/quota"
- "openshift.io/route"
- "openshift.io/security"
- "openshift.io/template"
```
reflecting the API groups. Though, it is still practical to link https://github.com/openshift/openshift-apiserver/blob/master/pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go (yet to be updated, still part of WIP PR) where users can find the list.

Reply:
> This type does not exist in kube-apiserver. It's used only for (de)serialization and meant exclusively for the openshift-apiserver and its operator.

What do you mean it's not there? Is this a config file format rather than a CR or aggregated API?

> I was hoping to reduce the usage only for the internal purposes. Is there any convention for internal use only purposes?

That's a good question and not something I'm familiar with, not sure what the rules are for "internal APIs", or whether we truly have internal APIs in openshift, I guess that depends on the answer to my previous question.

Linking to a source for valid options works for me I think

Reply:
Yeap. Config file format only.

Reply:
```
$ oc explain openshiftcontrolplane
the server doesn't have a resource type "openshiftcontrolplane"
```